app/src/main/java/com/example/playlistmaker/SearchActivity.kt

```diff
@@ -1,198 +1,200 @@
+        super.onRestoreInstanceState(savedInstanceState)
+        val searchText = savedInstanceState.getString(SEARCH_TEXT)
+        findViewById<EditText>(R.id.inputEditText).setText(searchText)
+    }
 
-class TrackViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
+    companion object {
+        const val SEARCH_TEXT = "searchText"
+    }
 
-    private val trackNameView: TextView
-    private val artistNameView: TextView
-    private val trackTimeView: TextView
-    private val artworkUrl100View: ImageView
+    private fun showPlaceholderError() {
+        trackListClear()
+        nothingWasFound.visibility = View.VISIBLE
+        placeholderNoInternet.visibility = View.GONE
+    }
 
-    init {
-        trackNameView = itemView.findViewById(R.id.trackName)
-        artistNameView = itemView.findViewById(R.id.artistName)
-        trackTimeView = itemView.findViewById(R.id.trackTime)
-        artworkUrl100View = itemView.findViewById(R.id.artworkUrl100)
+    private fun showPlaceholderForEmpty() {
+        trackListClear()
+        placeholderNoInternet.visibility = View.VISIBLE
+        nothingWasFound.visibility = View.GONE
     }
 
-    fun bind(track: Track) {
-        trackNameView.text = track.trackName
-        artistNameView.text = track.artistName
-        trackTimeView.text = track.trackTime
-        Glide.with(itemView.context)
-            .load(track.artworkUrl100)
-            .placeholder(R.drawable.placeholder)
-            .transform(RoundedCorners(2))
-            .into(artworkUrl100View)
+    private fun closePlaceholder() {
+        placeholderNoInternet.visibility = View.GONE
+        nothingWasFound.visibility = View.GONE
     }
-}
 
-class TrackAdapter(
-    private val tracksList: ArrayList<Track>,
-) : RecyclerView.Adapter<TrackViewHolder>() {
+    private fun searchTextRequest() {
+        if (searchEditText.text.isNotEmpty()) {
+            api.search(searchEditText.text.toString())
+                .enqueue(object : Callback<TrackResult> {
+                    @SuppressLint("NotifyDataSetChanged")
+                    override fun onResponse(
+                        call: Call<TrackResult>, response: Response<TrackResult>
+                    ) {
 
-    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TrackViewHolder {
-        val view = LayoutInflater.from(parent.context).inflate(R.layout.track_item, parent, false)
-        return TrackViewHolder(view)
-    }
 
-    override fun onBindViewHolder(holder: TrackViewHolder, position: Int) {
-        holder.bind(tracksList[position])
+                        if (response.code() == 200) {
+
+                            trackListClear()
+                            closePlaceholder()
+
+                            if (response.body()?.results?.isNotEmpty() == true) {
+                                trackList.addAll(response.body()?.results!!)
+                                adapter.notifyDataSetChanged()
```

Comment:
🍏 Для повышения читаемости рекомендую создать объявить отдельную переменную. А smart cast избавить нас от необходиости использовать  double bang:
```suggestion
                           val results = response.body()?.results
                            if (results?.isNotEmpty() == true) {
                                trackList.addAll(results)
                                adapter.notifyDataSetChanged()
```

https://kotlinlang.org/docs/typecasts.html#smart-casts